app/models/user.rb

```diff
@@ -221,6 +252,9 @@ def sync_profile
   before_create :set_default_language
   before_destroy :unsubscribe_from_newsletters, prepend: true
   before_destroy :destroy_follows, prepend: true
+
+  # NOTE: @citizen428 Temporarily while migrating to generalized profiles
+  before_save { |user| user.profile&.save if user.profile&.changed? }
```

Comment:
This is a bit of a hack but seems to work well: if we assign a new value to a delegated attribute and call `user.save`, the updated value will end up inside the profile's `data` attribute, so `profile.changed?` will be `true`.  I'm now wondering if I should change the condition to `valid? && user.profile&.changed? ` 🤔